runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/GrpcWindmillServer.java

```diff
@@ -1441,6 +1623,11 @@ private void issueSingleRequest(final long id, PendingRequest pendingRequest) {
           .setRequestId(id)
           .setShardingKey(pendingRequest.request.getShardingKey())
           .setSerializedWorkItemCommit(pendingRequest.request.toByteString());
+      if (!pendingRequest.latencyAttributions.isEmpty()) {
+        requestBuilder
+            .getCommitChunkBuilder(0)
+            .addAllPerWorkItemLatencyAttributions(pendingRequest.latencyAttributions);
```

Comment:
Let's consider adding the latency information for the work item itself to the WorkItemCommitRequest

Then we can just have similar timing information on the StreamingCommitWorkRequest to calculate latencies on the commit path.

Also note below you are not setting the latency information in issueMultiChunkRequest.

Reply:
Done. thanks for the suggestion.